lib/hive/controller/android.rb

```diff
@@ -6,107 +6,105 @@ module Hive
   class Controller
     class Android < Controller
 
-      # Uses either DeviceAPI or DeviceDB to generate queue names for a device
-      def calculate_queue_names(device)
-        if device.is_a? DeviceAPI::Android::Device
-          queues = [
-              device.model,
-              device.manufacturer,
-              'android',
-              "android-#{device.version}",
-              "android-#{device.version}-#{device.model}"
-          ]
-        else
-
-          queues = [
-              device['device_model'],
-              device['device_brand'],
-              device['os'],
-              "#{device['os']}-#{device['os_version']}",
-              "#{device['os']}-#{device['os_version']}-#{device['device_model']}"
-          ]
-
-          queues << device["features"] unless device["features"].empty?
-
-          queues.flatten
-        end
-        queues
+      def detect
+        self.detect_hive_mind
+        self.detect_devicedb
       end
 
-      def populate_queues(device)
-        queues = calculate_queue_names(device)
-
-        # Add the queue prefix if it has been setup in the config
-        queues = queues.map { |a| "#{@config['queue_prefix']}-#{a}"} if @config['queue_prefix']
-
-        devicedb_queues = device['device_queues'].map { |d| d['name'] }
-        # Check to see if the queues have already been registered with this device
-        missing_queues = (queues - devicedb_queues) + (devicedb_queues - queues)
-        return if missing_queues.empty?
-
-        queues << missing_queues
-
-        queue_ids = queues.flatten.uniq.map { |queue| find_or_create_queue(queue) }
-
-        values = {
-            name: device['name'],
-            hive_id: device['hive_id'],
-            feature_list: device['features'],
-            device_queue_ids: queue_ids
-        }
-
-        Hive.devicedb('Device').edit(device['id'], values)
-      end
+      # Register with Hive Mind (New)
+      def detect_hive_mind
+        devices = DeviceAPI::Android.devices
+        Hive.logger.debug('HM) No devices attached') if devices.empty?
+
+        if not Hive.hive_mind.device_details.has_key? :error
+          # Checking device_type for 'Mobile'
+          # TODO Check for os to ensure Android
+          connected_devices = Hive.hive_mind.device_details['connected_devices'].select{ |d| d['device_type'] == 'Mobile' }
```

Comment:
How is this going to work if a device has both iOS and Android devices attached to it?